src/os/layer/preset/layerpresetmanager.js

```diff
@@ -1,223 +1,240 @@
-        return presets;
-      }, this.handleLoadError, this);
+  /**
+   * @param {boolean|undefined} b
+   * @return {!boolean}
+   */
+  isAdmin(b) {
+    if (b === true || b === false) {
+      this.admin_ = (b === true);
     }
+    return this.admin_;
   }
-};
 
+  /**
+   * @param {string} key
+   * @param {IPresetService} service
+   * @param {...} opt
+   * @return {boolean} true if prior value overwritten
+   */
+  registerService(key, service, ...opt) {
+    this.presets_ = {}; // set presets to reload every layer
+    return this.services_.register.apply(this.services_, [key, service].concat(opt));
+  }
 
-/**
- * Handler for errors in loading presets.
- *
- * @param {*} reason
- */
-os.layer.preset.LayerPresetManager.prototype.handleLoadError = function(reason) {
-  var msg = 'Unspecified error.';
-  if (typeof reason == 'string') {
-    msg = reason;
-  } else if (reason instanceof Error) {
-    msg = reason.message;
+  /**
+   * @param {string} key
+   * @return {boolean} true if prior value dropped
+   */
+  unregisterService(key) {
+    this.presets_ = {}; // set presets to reload every layer
+    return this.services_.remove(key);
   }
 
-  goog.log.error(os.layer.preset.LayerPresetManager.LOGGER_, 'Failed to load presets. Reason: ' + msg);
-};
+  /**
+   * Gets a promise that resolves to the presets for a given layer type.
+   *
+   * @param {string} type
+   * @param {string} url
+   * @deprecated Please create a os.layer.preset.PresetService instead
+   */
+  registerPreset(type, url) {}
 
+  /**
+   * Handler for errors in loading presets.
+   *
+   * @param {*} reason
+   * @deprecated
+   */
+  handleLoadError(reason) {}
 
-/**
- * Gets a promise that resolves to the presets for a given layer ID.
- *
- * @param {string} id The layer ID.
- * @param {boolean=} opt_applyDefault Whether to apply the default styles on load.
- * @return {goog.Promise<Array<osx.layer.Preset>>|undefined}
- */
-os.layer.preset.LayerPresetManager.prototype.getPresets = function(id, opt_applyDefault) {
-  if (!this.presets_[id]) {
-    this.initPreset(id, opt_applyDefault);
+  /**
+   * Gets a promise that resolves to the presets for a given layer ID.
+   *
+   * @param {string} id The layer ID.
+   * @param {boolean=} opt_applyDefault Whether to apply the default styles on load.
+   * @return {Promise<Array<osx.layer.Preset>>|undefined}
+   */
+  getPresets(id, opt_applyDefault) {
+    if (!this.presets_[id]) {
+      this.initPreset(id, opt_applyDefault);
+    }
+
+    return this.presets_[id];
   }
 
-  return this.presets_[id];
-};
+  /**
+   * Initializes the layer presets for a layer.
+   *
+   * @param {string} id The layer ID.
+   * @param {boolean=} opt_applyDefault Whether to apply the default styles on load.
+   * @protected
+   */
+  initPreset(id, opt_applyDefault) {
+    // use the filter key to pull the value from settings
+    var filterKey;
+    var layer = os.map.mapContainer.getLayer(id);
```

Comment:
Sine you ES6-ified this class, references like this should be cleaned up as well.

Reply:
Doing goog.require('os.MapContainer') properly creates a circular dependency somewhere in the os.layer chain. Added a comment here; but pushing forward in the interest of completing this feature on time.